packages/NetteToSymfony/src/Rector/ClassMethod/RenameEventNamesInEventSubscriberRector.php

```diff
@@ -0,0 +1,197 @@
+            'TERMINATE',
+        ],
+        'Contributte\Events\Extra\Event\RequestEvent::NAME' => ['Symfony\Component\HttpKernel\KernelEvents', 'REQUEST'],
+        'Contributte\Events\Extra\Event\PresenterEvent::NAME' => [
+            'Symfony\Component\HttpKernel\KernelEvents',
+            'CONTROLLER',
+        ],
+        'Contributte\Events\Extra\Event\PresenterStartupEvent::NAME' => [
+            'Symfony\Component\HttpKernel\KernelEvents',
+            'CONTROLLER',
+        ],
+        'Contributte\Events\Extra\Event\PresenterShutdownEvent::NAME' => [
+            'Symfony\Component\HttpKernel\KernelEvents',
+            'CONTROLLER',
+        ],
+        'Contributte\Events\Extra\Event\ResponseEvent::NAME' => [
+            'Symfony\Component\HttpKernel\KernelEvents',
+            'RESPONSE',
+        ],
+        'Contributte\Events\Extra\Event\ErrorEvent::NAME' => ['Symfony\Component\HttpKernel\KernelEvents', 'EXCEPTION'],
+    ];
+
+    /**
+     * @var BetterNodeFinder
+     */
+    private $betterNodeFinder;
+
+    public function __construct(BetterNodeFinder $betterNodeFinder)
+    {
+        $this->betterNodeFinder = $betterNodeFinder;
+    }
+
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition('Changes event names from Nette ones to Symfony ones', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final class SomeClass implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return ['nette.application' => 'someMethod'];
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final class SomeClass implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [\SymfonyEvents::KERNEL => 'someMethod'];
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $classNode = $node->getAttribute(Attribute::CLASS_NODE);
+        if ($classNode === null) {
+            return null;
+        }
+
+        if (! $this->isType($classNode, 'Symfony\Component\EventDispatcher\EventSubscriberInterface')) {
+            return null;
+        }
+
+        if (! $this->isName($node, 'getSubscribedEvents')) {
+            return null;
+        }
+
+        /** @var Return_[] $returnNodes */
+        $returnNodes = $this->betterNodeFinder->findInstanceOf($node, Return_::class);
+
+        foreach ($returnNodes as $returnNode) {
+            if (! $returnNode->expr instanceof Array_) {
+                continue;
+            }
+
+            $this->renameArrayKeys($returnNode);
```

Comment:
I smell a bug here, that should be covered with test.

There are 2 possible usecases:
```php
public static function getSubscribedEvents(): array
	{
		return [StartupEvent::NAME => 'addHeaders'];
	}
```
and
```php
public static function getSubscribedEvents(): array
	{
		return [StartupEvent::NAME];
	}
```

Both are valid event subscribers. In 1st case, this works, because you rename array keys. For 2nd usecase this will not work.

Reply:
Resolved in person